src/store/slice/auth.slice.js

```diff
@@ -18,12 +20,24 @@ export const authSlice = createSlice({
 		},
 		toggleLoading: (state, action) => {
 			return { ...state, loading: action.payload };
+		},
+		toggleDarkMode: (state, action) => {
```

Comment:
reducers should always be pure functions with no side effects. Please move all the side effects into some other helper function which will dispatch actions to store. Ideally, `toggleDarkMode` should only be toggling the `state.darkMode` to `true` or `false`.